lib/utils.ts

```diff
@@ -464,6 +472,15 @@ export function isID(str: string): boolean {
   return x <= 2.2;
 }
 
+export function isUrl(str: string): boolean {
+  try {
+    new URL(str);
+    return true;
+  } catch {
+    return false;
+  }
+}
```

Comment:
This returns false if http is missing, for example www.google.com, do all the urls in the SEC example have a protocol?

Reply:
They do, but in general this is an assumption we're making. A simple response would be to add a `https://` if it doesn't start with one, but then email addresses count as URLs and so does any alphanumeric string, so that's not ideal either.

Will instead use a url regex. Think that'll do a better job